packages/cactus-plugin-persistence-fabric/src/main/typescript/plugin-persistence-fabric.ts

```diff
@@ -0,0 +1,914 @@
+    let tempBlockNumber = 0;
+    let blockNumber = tempBlockNumber.toString();
+    let block: AxiosResponse<GetBlockResponseV1> = {
+      data: { decodedBlock: {} },
+      status: 0,
+      statusText: "string",
+      headers: undefined,
+      config: {},
+      request: undefined,
+    };
+    let moreBlocks = true;
+    do {
+      try {
+        block = await this.apiClient.getBlockV1({
+          channelName: this.ledgerChannelName,
+          gatewayOptions: this.gatewayOptions,
+          query: {
+            blockNumber,
+          },
+        });
+      } catch (error) {
+        this.log.info("Last block in ledger", tempBlockNumber - 1);
+        moreBlocks = false;
+      }
+
+      if (block.status == 200) {
+        const logBlock = JSON.stringify(block.data);
+
+        // Put scrapped block into database
+        this.log.info(logBlock);
+        this.log.info("Block number: ", blockNumber);
+
+        // if (!this.dbClient.isConnected) {
+        //   await this.dbClient.connect();
+        //   this.isConnected = true;
+        // }
+        if (moreBlocks) {
+          await this.migrateBlockNrWithTransactions(blockNumber);
+          this.lastSeenBlock = tempBlockNumber;
+          this.lastBlock = tempBlockNumber;
+          tempBlockNumber = tempBlockNumber + 1;
+          blockNumber = tempBlockNumber.toString();
+          if (tempBlockNumber > edgeOfLedger) {
+            moreBlocks = false;
+          }
+        }
+      } else {
+        moreBlocks = false;
+      }
+    } while (moreBlocks);
+    return "done";
+  }
+
+  /**  Synchronization of blocks
+   * - Synchronize entire ledger state
+   * with the database as far as the lastBlockInLedger shows ( triggered once )
+   * @returns string promise lastBlock number after finishing the process
+   *
+   */
+
+  async continueBlocksSynchronization(
+    signingCredential: FabricSigningCredential,
+  ): Promise<string> {
+    this.lastSeenBlock = await this.dbClient.getMaxBlockNumber();
+    let tempBlockNumber = this.lastSeenBlock;
+    let blockNumber = tempBlockNumber.toString();
+    this.lastBlock = await this.lastBlockInLedger(signingCredential);
+    let block: AxiosResponse<GetBlockResponseV1> = {
+      data: { decodedBlock: {} },
+      status: 0,
+      statusText: "",
+      headers: undefined,
+      config: {},
+      request: undefined,
+    };
+    this.synchronizationGo = true;
+    do {
+      if (this.lastBlock > this.lastSeenBlock) {
+        try {
+          block = await this.apiClient.getBlockV1({
+            channelName: this.ledgerChannelName,
+            gatewayOptions: this.gatewayOptions,
+            query: {
+              blockNumber,
+            },
+          });
+        } catch (error) {
+          this.log.info("Last block in ledger", tempBlockNumber - 1);
+        }
+
+        if (block.status == 200) {
+          const tempBlockstep1 = JSON.stringify(block.data);
```

Comment:
Can we straightaway get the `tempBlock` var by executing JSON.parse(JSON.stringify(block.data))?

Reply:
ok removing

Reply:
ok removing